src/snowflake/connector/util_text.py

```diff
@@ -33,6 +43,109 @@ def split_statements(
     buf: StringIO,
     remove_comments: bool = False,
     delimiter: Optional[SQLDelimiter] = None,
+):
+    """Entry function to split statements."""
+    if delimiter is None or str(delimiter) == ";":
```

Comment:
> `str(delimiter) == ";"`

I think this is going to lead to bunch of unintended changes.

I think `SQLDelimiter` is only used by SnowSQL and so we should be able to get rid of the old code-path.